packages/strapi-plugin-content-manager/admin/src/components/RelationPreviewList/RelationPreviewTooltip.js

```diff
@@ -0,0 +1,91 @@
+import React, { useState, useEffect, useCallback, useRef, useLayoutEffect } from 'react';
+import { Text, Padded } from '@buffetjs/core';
+import { LoadingIndicator, request } from 'strapi-helper-plugin';
+import PropTypes from 'prop-types';
+
+import getRequestUrl from '../../utils/getRequestUrl';
+import getMockData from './mockData';
+import Tooltip from '../Tooltip';
+
+const RelationPreviewTooltip = ({ tooltipId, rowId, mainField, name }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [relationData, setRelationData] = useState([]);
+  const tooltipRef = useRef();
+  const abortController = new AbortController();
```

Comment:
Can you declare this constant inside your effect and pass the signal to the `fetchReflationData` so you don't have to pass the signal as a dependency of your effect?